plugins/repository-s3/src/main/java/org/elasticsearch/plugin/repository/s3/S3RepositoryPlugin.java

```diff
@@ -144,6 +145,7 @@ public void onModule(SettingsModule settingsModule) {
         settingsModule.registerSetting(S3Repository.Repository.STORAGE_CLASS_SETTING);
         settingsModule.registerSetting(S3Repository.Repository.CANNED_ACL_SETTING);
         settingsModule.registerSetting(S3Repository.Repository.BASE_PATH_SETTING);
+        settingsModule.registerSetting(S3Repository.Repository.THROTTLE_RETRIES_SETTING);
```

Comment:
Actually I think I have to revert this change.

At the very least I need to keep `KEY_SETTING` and `SECRET_SETTING`. Otherwise they are not masked when someone runs a `GET _snapshot/reponame`.

Test https://github.com/elastic/elasticsearch/blob/master/plugins/repository-s3/src/test/resources/rest-api-spec/test/repository_s3/20_repository.yaml fails then.

WDYT?